### PR TITLE
Add codes starting with 51

### PR DESCRIPTION
**Pull request by renetapopova**

### CODE REVIEW

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-status-codes-198.surge.sh

Updated pages:
- [List of all GQLSTATUS server error codes](https://neo4j-docs-status-codes-198.surge.sh/status-codes/5/errors/gql-errors)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-status-codes-198.surge.sh

Updated pages:
- [List of all GQLSTATUS server error codes](https://neo4j-docs-status-codes-198.surge.sh/status-codes/5/errors/gql-errors)

File: modules/ROOT/pages/errors/gql-errors.adoc [hunk 1]
```diff
@@ -432,3 +432,265 @@ Status description:: error: syntax error or access rule violation - auth info ex
...
 
 Status description:: error: syntax error or access rule violation - permission/access denied. Access denied, see the security logs for details.
+
+== System configuration or operation exception
+
+System configuration or operation exception errors occur when there is an error in the system configuration or operation, such as procedure registration failure, a missing class field annotation, an unsupported injectable component type, duplicate field names, invalid map key type, etc.
+
+=== 51N00
+
+Status description:: error: system configuration or operation exception - procedure registration error. Failed to register procedure.
```
Comment (Lojjs) [inline modules/ROOT/pages/errors/gql-errors.adoc:442]:
```suggestion
Status description:: error: system configuration or operation exception - procedure registration error. Failed to register procedure/function.
```

This seems to have been updated recently in the code base by @anderson4j, but it still say "Failed to register procedure." in the spreadsheet. Is the code variant correct?

File: modules/ROOT/pages/errors/gql-errors.adoc [hunk 2]
```diff
@@ -432,3 +432,265 @@ Status description:: error: syntax error or access rule violation - auth info ex
...
+
+
+=== 51N06
+
+Status description:: error: system configuration or operation exception - missing argument name. The argument at position `{ $pos }` in `{ $procMethod }` requires a '@Name' annotation and a non-empty name.
+
+
+=== 51N07
+
+Status description:: error: system configuration or operation exception - invalid ordering of default arguments. The `{ $procFun }` contains a non-default argument before a default argument. Non-default arguments are not allowed to be positioned after default arguments.
```
Comment (Lojjs) [inline modules/ROOT/pages/errors/gql-errors.adoc:476]: Although this match what is in the spreadsheet and codebase, I think the sentence is logically wrong. It says the non-default argument is _before_ a default argument but it is not allowed to be _after_ the default argument. I think both should be either before or after for it to make sense. Maybe @gem-neo4j can help us with what is correct here or tell me if I am just confused.

File: modules/ROOT/pages/errors/gql-errors.adoc [hunk 3]
```diff
@@ -432,3 +432,265 @@ Status description:: error: syntax error or access rule violation - auth info ex
...
+
+
+=== 51N41
+
+Status description:: error: system configuration or operation exception - admin operation failed. Server or database admin operation not possible.
+
+
+=== 51N43
+
+Status description:: error: system configuration or operation exception - cannot deallocate servers. Cannot deallocate server(s) `{ $serverAndedList }`.
```
Comment (Lojjs) [inline modules/ROOT/pages/errors/gql-errors.adoc:596]:
```suggestion
Status description:: error: system configuration or operation exception - cannot deallocate servers. Cannot deallocate server(s) `{ $serverList }`.
```

File: modules/ROOT/pages/errors/gql-errors.adoc [hunk 4]
```diff
@@ -432,3 +432,265 @@ Status description:: error: syntax error or access rule violation - auth info ex
...
+=== 51N52
+
+Status description:: error: system configuration or operation exception - cannot alter database topology. Number of primaries `{ $count }` needs to be at least 1 and may not exceed `{ $upper }`.
+
+
+=== 51N53
+
+Status description:: error: system configuration or operation exception - cannot alter database topology. Number of secondaries `{ $count }` needs to be at least 0 and may not exceed `{ $upper }`.
+
+
```
Comment (Lojjs) [inline modules/ROOT/pages/errors/gql-errors.adoc:648]:
```suggestion
=== 51N52

Status description:: error: system configuration or operation exception - number of primaries out of range. Cannot alter database topology.  Number of primaries `{ $count }` needs to be at least 1 and may not exceed `{ $upper }`.


=== 51N53

Status description:: error: system configuration or operation exception - number of secondaries out of range. Cannot alter database topology. Number of secondaries `{ $count }` needs to be at least 0 and may not exceed `{ $upper }`.

```

We noticed that these two had the same subcondition so they got modified a bit. I have updated the spreadsheet too now.

**Review by Lojjs — CHANGES_REQUESTED** [review]
Found four smaller differences compared with the codebase on this one

File: modules/ROOT/pages/errors/gql-errors.adoc (hunk 2 above)
Comment (renetapopova) [reply modules/ROOT/pages/errors/gql-errors.adoc:476]: Yes, this seems wrong.

File: modules/ROOT/pages/errors/gql-errors.adoc [hunk 5]
```diff
@@ -432,3 +432,264 @@ Status description:: error: syntax error or access rule violation - auth info ex
...
+=== 51N06
+
+Status description:: error: system configuration or operation exception - missing argument name. The argument at position `{ $pos }` in `{ $procMethod }` requires a '@Name' annotation and a non-empty name.
+
+
+=== 51N07
+
+Status description:: error: system configuration or operation exception - invalid ordering of default arguments. The `{ $procFun }` contains a non-default argument before a default argument. Non-default arguments are not allowed to be positioned after default arguments.
```
Comment (renetapopova) [inline modules/ROOT/pages/errors/gql-errors.adoc:476]:
```suggestion
Status description:: error: system configuration or operation exception - invalid ordering of default arguments. The `{ $procFun }` contains a non-default argument before a default argument. Non-default arguments are not allowed to be positioned before default arguments.
```

Comment (Lojjs) [reply modules/ROOT/pages/errors/gql-errors.adoc:476]: I check with Gem and it is the other way around. The first sentence is wrong and the second correct so we can have 

```suggestion
Status description:: error: system configuration or operation exception - invalid ordering of default arguments. The `{ $procFun }` contains a non-default argument after a default argument. Non-default arguments are not allowed to be positioned after default arguments.
```

or maybe

```suggestion
Status description:: error: system configuration or operation exception - invalid ordering of default arguments. The `{ $procFun }` contains a default argument before a non-default argument. Non-default arguments are not allowed to be positioned after default arguments.
```

**Review by Lojjs — APPROVED** [review]
Looks good now, thanks

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-status-codes-198.surge.sh

Updated pages:
- [List of all GQLSTATUS server error codes](https://neo4j-docs-status-codes-198.surge.sh/status-codes/5/errors/gql-errors)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-status-codes-198.surge.sh

Updated pages:
- [List of all GQLSTATUS server error codes](https://neo4j-docs-status-codes-198.surge.sh/status-codes/5/errors/gql-errors)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-status-codes-198.surge.sh

Updated pages:
- [List of all GQLSTATUS server error codes](https://neo4j-docs-status-codes-198.surge.sh/status-codes/5/errors/gql-errors)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-status-codes-198.surge.sh

Updated pages:
- [List of all GQLSTATUS server error codes](https://neo4j-docs-status-codes-198.surge.sh/status-codes/5/errors/gql-errors)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-status-codes-198.surge.sh

Updated pages:
- [List of all GQLSTATUS server error codes](https://neo4j-docs-status-codes-198.surge.sh/status-codes/5/errors/gql-errors)

**Pull request merged**